xwayland/xwm.c

```diff
@@ -1002,6 +1002,33 @@ static void xwm_handle_xcb_error(struct wlr_xwm *xwm, xcb_value_error_t *ev) {
 
 }
 
+static void xwm_handle_unhandled_event(struct wlr_xwm *xwm, xcb_generic_event_t *ev) {
+#ifdef WLR_HAS_XCB_ERRORS
+	xcb_errors_context_t *err_ctx;
+	if (xcb_errors_context_new(xwm->xcb_conn, &err_ctx)) {
+		wlr_log(L_DEBUG, "Could not allocate context for unhandled X11 event: %u",
+			ev->response_type);
+		return;
+	}
+
+	const char *extension;
+	const char *event_name =
+		xcb_errors_get_name_for_xcb_event(err_ctx, ev, &extension);
+	if (!event_name) {
+		wlr_log(L_DEBUG, "no name for unhandled event: %u",
+			ev->response_type);
+		xcb_errors_context_free(err_ctx);
+		return;
+	}
+
+	wlr_log(L_DEBUG, "unhandled X11 event: %s (%u)", event_name, ev->response_type);
```

Comment:
Why not also print the name of the extension that generated the event (`extension`)? (if one is provided, I'd guess this is `NULL` for core X11 events)

Reply:
I just missed it, feel free to add it -- my guess is extension will be null here 99% of the time but will be glad to test any PR you send us.